kolibri/utils/options.py

```diff
@@ -567,6 +572,82 @@ def get_configspec():
     return ConfigObj(lines, _inspec=True)
 
 
+def _set_from_envvars(conf):
+    """
+    Set the configuration from environment variables.
+    """
+    logger = _get_logger()
+    # keep track of which options were overridden using environment variables, to support error reporting
+    using_env_vars = {}
+
+    deprecation_warning = "Option {optname} in section [{section}] being overridden by deprecated environment variable {envvar}, please update to: {envvars}"
+    # override any values from their environment variables (if set)
+    # and check for use of deprecated environment variables and options
+    for section, opts in option_spec.items():
+        for optname, attrs in opts.items():
+            for envvar in attrs.get("envvars", []):
+                if os.environ.get(envvar):
+                    deprecated_envvars = attrs.get("deprecated_envvars", tuple())
+                    if envvar in deprecated_envvars:
+                        logger.warn(
+                            deprecation_warning.format(
+                                optname=optname,
+                                section=section,
+                                envvar=envvar,
+                                envvars=", ".join(
+                                    e
+                                    for e in attrs.get("envvars", [])
+                                    if e not in deprecated_envvars
+                                ),
+                            )
+                        )
+                    else:
+                        logger.info(
+                            "Option {optname} in section [{section}] being overridden by environment variable {envvar}".format(
+                                optname=optname, section=section, envvar=envvar
+                            )
+                        )
+                    if attrs.get("deprecated", False):
+                        logger.warn(
+                            "Option {optname} in section [{section}] is deprecated, please remove it from your options.ini file".format(
+                                optname=optname, section=section
+                            )
+                        )
+                    conf[section][optname] = os.environ[envvar]
+                    using_env_vars[optname] = envvar
```

Comment:
re https://github.com/learningequality/kolibri/pull/8208/files/7898777748b934fd2b9d401fb9a6c4c67c1c2696#r670551691

something like this:

```suggestion
def _set_from_envvar(conf, section, optname, attrs, envvar):
    deprecated_envvars = attrs.get("deprecated_envvars", tuple())
    if envvar in deprecated_envvars:
        logger.warn(
            deprecation_warning.format(
                optname=optname,
                section=section,
                envvar=envvar,
                envvars=", ".join(
                    e
                    for e in attrs.get("envvars", [])
                    if e not in deprecated_envvars
                ),
            )
        )
    else:
        logger.info(
            "Option {optname} in section [{section}] being overridden by environment variable {envvar}".format(
                optname=optname, section=section, envvar=envvar
            )
        )
    if attrs.get("deprecated", False):
        logger.warn(
            "Option {optname} in section [{section}] is deprecated, please remove it from your options.ini file".format(
                optname=optname, section=section
            )
        )
    conf[section][optname] = os.environ[envvar]
    using_env_vars[optname] = envvar


def _set_from_envvars(conf):
    """
    Set the configuration from environment variables.
    """
    logger = _get_logger()
    # keep track of which options were overridden using environment variables, to support error reporting
    using_env_vars = {}

    deprecation_warning = "Option {optname} in section [{section}] being overridden by deprecated environment variable {envvar}, please update to: {envvars}"
    # override any values from their environment variables (if set)
    # and check for use of deprecated environment variables and options
    for section, opts in option_spec.items():
        for optname, attrs in opts.items():
            for envvar in attrs.get("envvars", []):
                if os.environ.get(envvar):
                    _set_from_envvar(conf, section, optname, attrs, envvar)
```